app/services/processors/blog_metrics_updater.rb

```diff
@@ -0,0 +1,52 @@
+module Processors
+  class BlogMetricsUpdater < BaseService
+    def call
+      update_blog_post_visits_metrics
+      update_technologies_visits_metrics
+    end
+
+    private
+
+    def update_blog_post_visits_metrics
+      BlogPost.find_each(batch_size: 50) do |blog_post|
+        BlogPostViewsUpdater.call(blog_post.blog_id)
+      end
+    end
+
+    def update_technologies_visits_metrics
+      blog_views_by_technology_and_month.each do |pair_group, visits|
+        timestamp, technology_id = pair_group
+        update_technology_visits_metric(technology_id, visits, timestamp)
+      end
+    end
+
+    def blog_views_by_technology_and_month
+      Metric.where(name: Metric.names[:blog_visits], ownable_type: BlogPost.to_s)
+            .where('value_timestamp >= ?', latest_technology_metrics_updated_at)
+            .joins('JOIN blog_posts on metrics.ownable_id = blog_posts.id')
+            .group(:value_timestamp, :technology_id)
+            .sum(:value)
+    end
+
+    def latest_technology_metrics_updated_at
+      latest_visits_metrics_by_technology =
```

Comment:
Can we memoize this? So that it doesn't run a query for each iteration.

Reply:
I could do it, absolutely, but I think it still runs just once.
It's used in `blog_views_by_technology_and_month`, and that method is called once to retrieve all the records. Then it just iterates on the results.
Or am I missing something?

Reply:
You're absolutely right, my bad.